v2/uci/src/modules/bot/bot.controller.spec.ts

```diff
@@ -1,20 +1,210 @@
+        },
+        "name": null
+      }
+    }
+  ],
+  "logicIDs": [
+    {
+      "id": "testLogicId",
+      "name": "Load Test Firebase Broadcast Logic",
+      "createdAt": new Date("2023-05-05T09:28:11.910Z"),
+      "updatedAt": new Date("2023-05-05T09:28:11.911Z"),
+      "description": null,
+      "adapterId": "testAdapterId",
+      "transformers": [
+        {
+          "id": "testTransformerId",
+          "createdAt": new Date("2023-05-05T09:28:11.894Z"),
+          "updatedAt": new Date("2023-05-05T09:28:11.912Z"),
+          "meta": {
+            "body": "Hello ${name}-${phoneNo}, Test Notification",
+            "type": "broadcast",
+            "title": "Firebase Test Notification",
+            "params": [
+              "name",
+              "phoneNo"
+            ],
+            "templateType": "JS_TEMPLATE_LITERALS"
+          },
+          "transformerId": "testTransformerId",
+          "conversationLogicId": "testConversationLogicId"
+        }
+      ],
+      "adapter": {
+        "id": "testAdapterId",
+        "createdAt": new Date("2023-03-18T06:02:41.824Z"),
+        "updatedAt": new Date("2023-03-18T06:02:41.824Z"),
+        "channel": "web",
+        "provider": "firebase",
+        "config": {
+          "credentials": {
+            "vault": "testVault",
+            "variable": "nl-app-firebase-notification"
+          }
+        },
+        "name": "Test Firebase Adapter"
+      }
+    }
+  ]
+};
 
 describe('BotController', () => {
-  let controller: BotController;
+  let botController: BotController;
+  let configService: ConfigService;
 
   beforeEach(async () => {
+    jest.setTimeout(30000);
     const module: TestingModule = await Test.createTestingModule({
       controllers: [BotController],
-      providers: [BotService],
+      providers: [
+        BotService,
+        GQLResolverService,
+        GetRequestResolverService,
+        PostRequestResolverService,
+        SecretsService,
+        ServiceService,
+        DeviceManagerService,
+        TelemetryService,
+        ConfigService, {
+          provide: ConfigService,
+          useClass: mockConfigService,
+        },
+        PrismaService, {
+          provide: PrismaService,
+          useClass: MockPrismaService,
+        }
+      ],
     }).compile();
 
-    controller = module.get<BotController>(BotController);
+    botController = module.get<BotController>(BotController);
+    configService = module.get<ConfigService>(ConfigService);
   });
 
-  it('should be defined', () => {
-    expect(controller).toBeDefined();
+  it('bot start passes admin token to segment url', async () => {
+    const headers = {'admin-token': 'testAuthToken'};
+    const botId = 'testBotId';
+    let submittedToken;
+    fetchMock.getOnce('http://testSegmentUrl/segments/1/mentors?deepLink=nipunlakshya://chatbot&limit=1&offset=0', (url, options) => {
+      if (options.headers) {
+        submittedToken = options.headers['admin-token'];
```

Comment:
Hey @ChakshuGautam, `SecretsService.getAllSecrets()` returns an array instead of an actual `Headers` object. This is further passed in the fetch request as is [here](https://github.com/samagra-comms/uci-apis/blob/6bfadeda1ac4755143f1e036b2da3dc8576ffafd/v2/uci/src/modules/service/http-get.resolver.ts#L36), which results in the headers field being an array instead of a `HeadersInit`. Hence I had to extract the headers differently in both the fetchMocks. Should this be fixed by converting the result returned by `SecretsService.getAllSecrets()`  into an actual `Headers` object before passing it to fetch?